modules/button/react/lib/parts/ButtonContainer.tsx

```diff
@@ -59,7 +59,14 @@ function getIconColorSelectors(
         fill: fill
           ? color === themePrimary.contrast
             ? themePrimary.main
-            : themePrimary.contrast
+            : themePrimary.contract
+          : color,
+      },
+      '.wd-icon-accent:nth-of-type(2), .wd-icon-accent2:nth-of-type(2)': {
```

Comment:
This seems to do the trick, but I'm not convinced of the solution, maybe we need to pass in an accent prop?